connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -3629,6 +3630,64 @@ public void testPollDurationOnSlowConnectorOperations() {
         PowerMock.verifyAll();
     }
 
+    @Test(expected = RejectedExecutionException.class)
+    @SuppressWarnings("unchecked")
+    public void shouldThrowWhenStartAndStopExecutorThrowsRejectedExecutionExceptionAndHerderNotStopping() throws InterruptedException {
+        ExecutorService startAndStopExecutor = EasyMock.mock(ExecutorService.class);
+        herder.startAndStopExecutor = startAndStopExecutor;
+
+        Callable<Void> connectorStartingCallable = () -> null;
+
+        EasyMock.expect(startAndStopExecutor.invokeAll(EasyMock.anyObject(Collection.class))).andThrow(new RejectedExecutionException());
+
+        PowerMock.replayAll(startAndStopExecutor);
+
+        herder.startAndStop(Collections.singletonList(connectorStartingCallable));
+
+    }
```

Comment:
I really like that we've added test coverage for this, but I think we can get better guarantees by testing with `Herder::tick` and using the herder's actual `startAndStopExecutor`, instead of mocking out the executor and calling `startAndStop` directly:

```suggestion
    @Test
    public void shouldThrowWhenStartAndStopExecutorThrowsRejectedExecutionExceptionAndHerderNotStopping() {
        EasyMock.expect(member.memberId()).andStubReturn("leader");
        expectRebalance(1, Arrays.asList(CONN1), Collections.emptyList(), true);
        expectConfigRefreshAndSnapshot(SNAPSHOT);
        EasyMock.expect(member.currentProtocolVersion()).andStubReturn(CONNECT_PROTOCOL_V0);

        PowerMock.replayAll();

        herder.startAndStopExecutor.shutdown();
        assertThrows(RejectedExecutionException.class, herder::tick);

        PowerMock.verifyAll();
    }
```